src/common/transformations/include/transformations/op_conversions/convert_pooling_to_reduce.hpp

```diff
@@ -0,0 +1,78 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <algorithm>
+#include <memory>
+#include <ngraph/op/util/op_types.hpp>
+#include <ngraph/opsets/opset1.hpp>
+#include <ngraph/pass/graph_rewrite.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/rt_info.hpp>
+#include <ngraph/validation_util.hpp>
+#include <transformations_visibility.hpp>
+#include <vector>
+
+namespace ngraph {
+namespace pass {
+
+class TRANSFORMATIONS_API ConvertPoolingToReduce;
+class TRANSFORMATIONS_API ConvertAvgPoolingToReduce;
+
+}  // namespace pass
+}  // namespace ngraph
+
+class ConvertPoolingBase : public ngraph::pass::MatcherPass {
+public:
+    template <class T>
+    ngraph::matcher_pass_callback convert_pooling_to_reduce();
+};
+
+class ngraph::pass::ConvertAvgPoolingToReduce : public ConvertPoolingBase {
+public:
+    OPENVINO_RTTI("ConvertAvgPoolingToReduce", "0");
+    ConvertAvgPoolingToReduce();
+};
+
+template <class T>
+ngraph::matcher_pass_callback ConvertPoolingBase::convert_pooling_to_reduce() {
```

Comment:
Please implement tests for transformation

Reply:
Where should I implement a test for this transformation? GPU plugin has no unit test for graph transformation. Do you mean functional test?